pkg/issuer/acme/dns/util/livedns_test.go

```diff
@@ -0,0 +1,164 @@
+//go:build livedns_test
+
+/*
+Copyright 2025 The cert-manager Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/*
+Addition to the above license statement:
+
+This file *may* contain code directly taken from the 'xenolf/lego' project.
+
+A copy of the license for this code can be found in the file named LICENSE in
+this directory.
+*/
+
+package util
+
+// The tests in this file connect to *live* DNS or DNS-over-HTTPS services,
+// and rely on various DNS records which are out of the control of the cert-manager
+// project. As such, these tests are:
+// 1. More likely to flake due to network connectivity issues
+// 2. Liable to break if the upstream DNS records change
+// 3. Unable to run in restrictive computing environments
+//    (such as MitM corporate proxies which block DNS / DNS-over-HTTPS)
+
+// Because of the above, these tests live behind a build tag so they're not
+// run by mistake.
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+)
+
+const (
+	standardTimeout = time.Second * 5
+)
```

Comment:
note: Most of these tests are lifted + shifted pretty directly, but I made a few changes:

1. Used a context with a timeout instead of `TODO`
2. Changed error messages (which referred to a domain used in an old test)
3. Used `t.Run` in the "table" tests (mostly to facilitate better use of `defer cancel()`)